### ISSUE TITLE
A11y_.Net Core_Winforms_MainForm_Common Controls1_Screen reader : Narrator is announcing the Time when the focus lands to the date picker control , but visually time is not present.

### ISSUE THREAD

**Issue by Ashutoshsahoo164** [dotnet/winforms#9232]
### .NET version

.NET-SDK-8.0.100-preview.4.23260.5

### Did it work in .NET Framework?

No

### GitHubTags:
#A11yMAS;#A11yTCS;#BM_.NET Core[Winforms]-Win32-May2023;#.NET Core;#A11ySev3;#WCAG1.3.1;#DesktopApp;#Win11;#Benchmark;#Narrator;#NVDA;

### Actual:
The Narrator screen reader is announcing the Time along with Date when the Focus lands on the Date control in the Date Picker section , But visually there is no Time stamp provided in the screen.

Note : This issue is also reproducing with NVDA screen reader. NVDA is announcing the "Effects" Label but not the "Sample" Label

### Expected:
The Narrator/NVDA screen reader should announce the contents which are present visually. The time stamp should be visible in the UI or Screen reader should not read the time stamp if it is not present visually.

### User Impact:
Low vision and Screen reader user will be confused with this announcement and content not present visually.

### Steps to reproduce

1. Install and Open Winforms App
2. Main Form screen would appear.
3. Tab till "Common Controls_1: Testing the controls under common controls Tab" and Hit enter.
4. Navigate to "Date picker" Section
5. Observe with Screen reader that the Time is announced by Narrator/NVDA but visually it is not present.

### Attachment : 

Narrator Attachment : 
![Narrator Time](https://github.com/dotnet/winforms/assets/85863895/c60ff7f4-e2e5-46fb-a7d2-8e5358cbff67)

NVDA Attachment:
![NVDA Timing](https://github.com/dotnet/winforms/assets/85863895/dd5e97b2-76a1-4473-8d62-3f173c8130c5)

**Comment by Amy-Li03** [comment]
This issue only reproduces on .NET 7.0 & 8.0, it doesn't repro on .NET 5.0 & 6.0 and .NET Framework apps.
**Test result on .NET 6.0:**
![image](https://github.com/dotnet/winforms/assets/62929087/0b60b7bb-82f6-478b-83d3-c36b5a9bd281)
**Test result on .NET 8.0:**
![image](https://github.com/dotnet/winforms/assets/62929087/8254dcee-0395-497b-ad83-7654bc9bddca)

**Comment by Ashutoshsahoo164** [comment]
Hi @Amy-Li03 I got , but the fix should be in latest .Net 8 version also

**Comment by merriemcgaw** [comment]
@Ashutoshsahoo164 thanks - @Amy-Li03 was confirming for us that we must have done something on our end to cause the date to be narrated. @LeafShi1 and @dmitrii-drobotov can you guys hunt down where we changed this?

**Comment by dmitrii-drobotov** [comment]
We should make Narrator read exactly what's displayed on the DateTimePicker, depending on `Format` and `CustomFormat` properties — it should read time when time is displayed, read full date as in .NET 6 when it's displayed in full format, etc. Currently it looks like there is only support for `CustomFormat` (added in https://github.com/dotnet/winforms/pull/6637):
 
![datetimepicker](https://github.com/dotnet/winforms/assets/102954094/d0ff3cca-bdd2-499c-9a88-87170ca12f0a)

And most likely like we regressed it from https://github.com/dotnet/winforms/pull/6620, when we added UIA provider for it, but didn't account for format.

**Comment by LeafShi1** [comment]
> @Ashutoshsahoo164 thanks - @Amy-Li03 was confirming for us that we must have done something on our end to cause the date to be narrated. @LeafShi1 and @dmitrii-drobotov can you guys hunt down where we changed this?

Epica is focused on this, so it is assigned to him.

**Comment by Nora-Zhou01** [comment]
Verified this issue in the latest .NET 8.0 SDK build: 8.0.100-preview.7.23323.19 on winforms repo, When the focus falls on the date picker control, the narrator announces what is visible to the user interface. Screenshot as below:
![1111111111111111](https://github.com/dotnet/winforms/assets/104609169/02694b6a-04e4-477c-aeb2-11464da8154d)

**Comment by MandiMan** [comment]
Verified this on .NET 8.0 TP build: 8.0.100-preview.6.23327.6, issue was fixed, result is same as above test result.
